repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/do-upgrade.sh

```diff
@@ -13,13 +13,55 @@ export LEAPP3_BIN=$LEAPPHOME/leapp3
 
 export NEWROOT=${NEWROOT:-"/sysroot"}
 
+export LEAPP_DEBUG_ROOT="/tmp/leapp-debug-root"
+export LEAPP_DEBUG_DATA_ROOT="${LEAPP_DEBUG_ROOT}/leapp-data"
+export LEAPP_DEBUG_DATA_JOURNALCTL="${LEAPP_DEBUG_DATA_ROOT}/journactl.log"
+export LEAPP_DEBUG_DATA_TARBALL="${LEAPP_DEBUG_ROOT}/leapp-debug.tar.xz"
+
+collect_and_dump_debug_data() {
+    # Collection
+    journalctl -amo verbose > $LEAPP_DEBUG_DATA_JOURNALCTL
+    if [ -f "${NEWROOT}/var/lib/leapp/leapp.db" ]; then
+        cp "${NEWROOT}/var/lib/leapp/leapp.db" "${LEAPP_DEBUG_DATA_ROOT}/leapp.db"
+    fi
+    if [ -d "${NEWROOT}/var/log/leapp" ]; then
+        cp -a "${NEWROOT}/var/log/leapp" "${LEAPP_DEBUG_DATA_ROOT}/leapp-logs"
+    fi
+    
+    # Tarball generation
+    tar -cJf $LEAPP_DEBUG_DATA_TARBALL $LEAPP_DEBUG_DATA_ROOT
+    
+    # Hashing
+    export LEAPP_DEBUG_DATA_TARBALL_MD5=$(md5sum $LEAPP_DEBUG_DATA_TARBALL)
+    export LEAPP_DEBUG_DATA_TARBALL_SHA256=$(sha256sum $LEAPP_DEBUG_DATA_TARBALL)
+    
+    # Data formatting
+    export LEAPP_DEBUG_DATA_TARBALL_BASE64=$(base64 -w 0 $LEAPP_DEBUG_DATA_TARBALL)
+    export LEAPP_DEBUG_DATA_TARBALL_BASE64_SIZE="${#LEAPP_DEBUG_DATA_TARBALL_BASE64}"
+    
+    export LEAPP_DEBUG_MESSAGE="@@@^^^@@@LEAPP_DEBUG_DATA@@@^^^@@@"
+    export LEAPP_DEBUG_MESSAGE="${LEAPP_DEBUG_MESSAGE}:MD5SUM=${LEAPP_DEBUG_DATA_TARBALL_MD5}"
+    export LEAPP_DEBUG_MESSAGE="${LEAPP_DEBUG_MESSAGE}:SHA256SUM=${LEAPP_DEBUG_DATA_TARBALL_SHA256}"
+    export LEAPP_DEBUG_MESSAGE="${LEAPP_DEBUG_MESSAGE}:B64LEN=${LEAPP_DEBUG_DATA_TARBALL_BASE64_SIZE}"
+    export LEAPP_DEBUG_MESSAGE="${LEAPP_DEBUG_MESSAGE}:DATA=${LEAPP_DEBUG_DATA_TARBALL_BASE64}"
+    export LEAPP_DEBUG_MESSAGE="${LEAPP_DEBUG_MESSAGE}:@@@^^^@@@LEAPP_DEBUG_DATA@@@^^^@@@"
+    
+    echo $LEAPP_DEBUG_MESSAGE
+    echo $LEAPP_DEBUG_MESSAGE
+    echo $LEAPP_DEBUG_MESSAGE
+    echo $LEAPP_DEBUG_MESSAGE
+    echo $LEAPP_DEBUG_MESSAGE
+}
+
+
+
 do_upgrade() {
     local args="" rv=0
     #FIXME: set here params we would like to possible use...
     #getargbool 0 rd.upgrade.test && args="$args --testing"
     #getargbool 0 rd.upgrade.verbose && args="$args --verbose"
     getargbool 0 rd.upgrade.debug && args="$args --debug"
-
+    
```

Comment:
a broken editor config?  (also lots of other places)

Reply:
Yeah I didn't even notice
But ignore this for now

I am less interested in a codestyle review . I need functionality feedback
We can tune the looks later, right now it's about feasibility because for me the amount of data printed was huge

You got to make leapp fail I. Initram on purpose however
For this I made the remove boot entry actor raise an exception after it is do e